azurerm/resource_arm_cosmos_db_account.go

```diff
@@ -229,31 +444,56 @@ func resourceArmCosmosDBAccountRead(d *schema.ResourceData, meta interface{}) er
 
 	d.Set("name", resp.Name)
 	d.Set("location", azureRMNormalizeLocation(*resp.Location))
-	d.Set("resource_group_name", resGroup)
+	d.Set("resource_group_name", resourceGroup)
+	flattenAndSetTags(d, resp.Tags)
+
 	d.Set("kind", string(resp.Kind))
 	d.Set("offer_type", string(resp.DatabaseAccountOfferType))
 	d.Set("ip_range_filter", resp.IPRangeFilter)
+	d.Set("endpoint", resp.DocumentEndpoint)
+
+	if v := resp.EnableAutomaticFailover; v != nil {
+		d.Set("enable_automatic_failover", resp.EnableAutomaticFailover)
+	}
+
 	flattenAndSetAzureRmCosmosDBAccountConsistencyPolicy(d, resp.ConsistencyPolicy)
-	flattenAndSetAzureRmCosmosDBAccountFailoverPolicy(d, resp.FailoverPolicies)
+	if _, ok := d.GetOk("failover_policy"); ok {
+		flattenAndSetAzureRmCosmosDBAccountFailoverPolicy(d, resp.FailoverPolicies)
```

Comment:
can we make this flatten and have the Set handled here? Unfortunately tags is a bad example that should be refactored at some point tbh